src/Microsoft.ML.TimeSeries/SrCnnTransformBase.cs

```diff
@@ -0,0 +1,319 @@
+            int backAddWindowSize, int lookaheadWindowSize, int averagingWindowSize, int judgementWindowSize, Double alertThreshold)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(name), windowSize, initialWindowSize, outputColumnName, inputColumnName, new VectorDataViewType(NumberDataViewType.Double, 3))
+        {
+            //TODO: Check argument
+
+            BackAddWindowSize = backAddWindowSize;
+            LookaheadWindowSize = lookaheadWindowSize;
+            AvergingWindowSize = averagingWindowSize;
+            JudgementWindowSize = judgementWindowSize;
+            AlertThreshold = alertThreshold;
+
+            OutputLength = 3;
+        }
+
+        private protected SrCnnTransformBase(IHostEnvironment env, ModelLoadContext ctx, string name)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(name), ctx)
+        {
+            OutputLength = 3;
+
+            byte temp;
+            temp = ctx.Reader.ReadByte();
+            BackAddWindowSize = (int)temp;
+            Host.CheckDecode(BackAddWindowSize > 0);
+
+            temp = ctx.Reader.ReadByte();
+            LookaheadWindowSize = (int)temp;
+            Host.CheckDecode(LookaheadWindowSize > 0);
+
+            temp = ctx.Reader.ReadByte();
+            AvergingWindowSize = (int)temp;
+            Host.CheckDecode(AvergingWindowSize > 0);
+
+            temp = ctx.Reader.ReadByte();
+            JudgementWindowSize = (int)temp;
+            Host.CheckDecode(JudgementWindowSize > 0);
+
+            temp = ctx.Reader.ReadByte();
+            AlertThreshold = (double)temp;
+            Host.CheckDecode(AlertThreshold >= 0 && AlertThreshold <= 1);
+        }
+
+        private protected SrCnnTransformBase(SrCnnArgumentBase args, string name, IHostEnvironment env)
+            : this(args.WindowSize, args.InitialWindowSize, args.Source, args.Name,
+                  name, env, args.BackAddWindowSize, args.LookaheadWindowSize, args.AvergingWindowSize, args.JudgementWindowSize, args.Threshold)
+        {
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+
+            Host.Assert(WindowSize > 0);
+            Host.Assert(InitialWindowSize == WindowSize);
+            Host.Assert(BackAddWindowSize > 0);
+            Host.Assert(LookaheadWindowSize > 0);
+            Host.Assert(AvergingWindowSize > 0);
+            Host.Assert(JudgementWindowSize > 0);
+            Host.Assert(AlertThreshold >= 0 && AlertThreshold <= 1);
+
+            base.SaveModel(ctx);
+            ctx.Writer.Write((byte)BackAddWindowSize);
+            ctx.Writer.Write((byte)LookaheadWindowSize);
+            ctx.Writer.Write((byte)AvergingWindowSize);
+            ctx.Writer.Write((byte)JudgementWindowSize);
+            ctx.Writer.Write((byte)AlertThreshold);
```

Comment:
`AlertThreshold` is a double - is casting to a `byte` going to round trip the value correctly?

Reply:
Fixed this bug. Thanks for pointing out!